ateam_ai/src/ateam_ai_node.cpp

```diff
@@ -163,6 +170,57 @@ class ATeamAINode : public rclcpp::Node
     ball_state.vel.y() = ball_state_msg->twist.linear.y;
   }
 
+  void field_callback(const ssl_league_msgs::msg::VisionGeometryFieldSize::SharedPtr field_msg)
+  {
+
+    Field field {
+        .field_length   = field_msg->field_length,
+        .field_width    = field_msg->field_width,
+        .goal_width     = field_msg->goal_width,
+        .goal_depth     = field_msg->goal_depth,
+        .boundary_width = field_msg->boundary_width
+    };
+
+    auto check_field_line_name = [](ssl_league_msgs::msg::VisionFieldLineSegment line_msg, std::string target_name) -> bool {
+        return line_msg.name == target_name; 
+    };
+    std::array<std::string, 4> field_bound_names = {"TopTouchLine", "BottomTouchLine", "LeftGoalLine", "RightGoalLine"};
+    for (size_t i = 0; i < field_bound_names.size(); i++) {
+        auto& name = field_bound_names.at(i);
+        auto itr = std::find_if(begin(field_msg->field_lines), end(field_msg->field_lines), std::bind(check_field_line_name, std::placeholders::_1, name));
+        if (itr != end(field_msg->field_lines)) {
+            field.field_corners.at(i)(0) = itr->p1.x;
+            field.field_corners.at(i)(1) = itr->p1.y;
+        }
+    }
+
+    FieldSidedInfo left_side_info {};
+    Eigen::Vector2d(-field.field_length/2.0, field.goal_width/2.0);
+    Eigen::Vector2d(-field.field_length/2.0, -field.goal_width/2.0);
+
+    FieldSidedInfo right_side_info {};
+    Eigen::Vector2d(field.field_length/2.0, field.goal_width/2.0);
+    Eigen::Vector2d(field.field_length/2.0, -field.goal_width/2.0);
+
+    //TODO assign based off known team info
+    field.ours = left_side_info;
+    field.theirs = right_side_info;
+
+    // Cant really find anything in the message that gives us goal position that isnt centered
+    // If people want more info will have to update in the future
+
+    // std::array<std::string, 4> left_bound_names = {"LeftFieldLeftPenaltyStretch", "LeftPenaltyStretch", "LeftFieldRightPenaltyStretch"}
+    // FieldSidedInfo left_side_info {};
+    // size_t i = 0;
+    // for (auto& name : field_bound_names) {
+
+    // }
+    //  std::array<std::string, 4> right_bound_names = {"RightFieldLeftPenaltyStretch", "RightPenaltyStretch", "RightFieldRightPenaltyStretch")
```

Comment:
I think I'm cool saying that if the goal isn't centered, there's some other issues going on. We can delete this and if it ever comes back up, we can look back at this PR/commit and see what we should have added

Reply:
Yeah thats kinda what I was thinking just thought at some point I might find some info on how to read that but doesnt look like it. That will give us the penalty box area. But the goal box and goal are things I couldnt find info on